src/middleware/return_direct.py

```
@@ -0,0 +1,35 @@
+
+class ReturnDirectMiddleware(AgentMiddleware[AgentState, AgentContext]):
+    """Middleware to handle return_direct behavior for tools.
+
+    Checks for:
+    1. Tools with return_direct=True attribute
+    2. ToolMessages with return_direct=True attribute (e.g., denied actions)
+    """
+
+    @hook_config(can_jump_to=["end"])
+    async def abefore_model(
+        self, state: AgentState, runtime: Runtime[AgentContext]
```

Ruff (0.14.3)

Unused method argument: `runtime`

(ARG002)